gui/src/main/window_behaviour.js

```diff
@@ -1,29 +1,104 @@
-const {shell} = require('electron')
+const {BrowserWindow, ipcMain, shell} = require('electron')
+const path = require('path')
 
-module.exports.openExternalLinks = (win) => {
-  win.webContents.on('will-navigate', (event, url) => {
-    if (url.startsWith('http') && !url.match('/auth/authorize')) {
-      event.preventDefault()
-      shell.openExternal(url)
-    }
-  })
-  return win
-}
+const ELMSTARTUP = 400
 
-module.exports.noMenu = (win) => {
-  win.setMenu(null)
-  win.setAutoHideMenuBar(true)
-}
+const log = require('cozy-desktop').default.logger({
+  component: 'windows'
+})
+
+module.exports = class WindowManager {
+  constructor (app, desktop) {
+    this.win = null
+    this.app = app
+    this.desktop = desktop
+    this.log = require('cozy-desktop').default.logger({
+      component: 'window' + this.windowOptions().title
+    })
 
-module.exports.devTools = (win) => {
-  if (process.env.WATCH === 'true' || process.env.DEBUG === 'true') {
-    win.webContents.openDevTools({mode: 'detach'})
+    let handlers = this.ipcEvents()
+    Object.keys(handlers).forEach((name) => {
+      if (!handlers[name]) {
+        throw new Error('undefined handler for event ' + name)
+      }
+      ipcMain.on(name, handlers[name])
+    })
   }
-}
 
-module.exports.dockApple = (app, win) => {
-  if (process.platform === 'darwin') {
-    app.dock.show()
-    win.on('closed', () => { app.dock.hide() })
+  /* abtract */
+  windowOptions () {
+    throw new Error('extends WindowManager before use')
+  }
+
+  /* abtract */
+  ipcEvents () {
+    throw new Error('extends WindowManager before use')
+  }
+
+  show () {
+    if (!this.win) return this.create()
+    this.win.show()
+    return Promise.resolve(this.win)
+  }
+
+  hide () {
+    if (this.win) this.win.close()
+    this.win = null
+  }
+
+  reload () {
+    if (this.win) this.win.reload()
+  }
+
+  send (...args) {
+    this.win && this.win.webContents && this.win.webContents.send(...args)
+  }
+
+  hash () {
+    return ''
+  }
+
+  create () {
+    this.win = new BrowserWindow(this.windowOptions())
+
+    // openExternalLinks
+    this.win.webContents.on('will-navigate', (event, url) => {
+      if (url.startsWith('http') && !url.match('/auth/authorize')) {
```

Comment:
Isn't `/auth/authorize` onboarding-specific?

Reply:
This is for all other links (`!url.match`)

Reply:
Oh, ok!